src/child.rs

```diff
@@ -0,0 +1,176 @@
+//! Utilties for managing child processes.
+//!
+//! This module helps us ensure that:
+//!
+//! * All child processes that we spawn get properly logged and their output is
+//!   logged as well.
+//!
+//! * That any "quick running" child process does not spam the console with its
+//!   output.
+//!
+//! * That any "long running" child process gets its output copied to our
+//!   stderr, so that the user isn't sitting there wondering if anything at all
+//!   is happening. This is important for showing `cargo build`'s output, for
+//!   example.
+
+use error::Error;
+use failure;
+use slog::Logger;
+use std::{
+    io::{self, BufRead},
+    process,
+    sync::mpsc,
+    thread, time,
+};
+use PBAR;
+
+fn taking_too_long(since: time::Instant) -> bool {
+    since.elapsed() > time::Duration::from_millis(200)
+}
+
+enum Output<S: AsRef<str>> {
+    Stdout(S),
+    Stderr(S),
+}
+
+fn print_child_output<S>(out: Result<Output<S>, io::Error>, command_name: &str)
+where
+    S: AsRef<str>,
+{
+    let message = match out {
+        Ok(Output::Stdout(line)) => format!("{} (stdout): {}", command_name, line.as_ref()),
+        Ok(Output::Stderr(line)) => format!("{} (stderr): {}", command_name, line.as_ref()),
```

Comment:
I'm not sure we want to prefix things with `(stderr)` and/or `(stdout)` as it may quickly add noise to the output. Do you think it's worth differentiating? (people may be alarmed, for example, that Cargo always prints to stderr despite it typically not generating errors).

Additionally, the `command_name` may be best to print at the very beginning? Something like `Running ...` like in Cargo so it's there if necessary and otherwise not taking up too much space.